lib/livebook/runtime/standalone.ex

```diff
@@ -131,88 +126,116 @@ defmodule Livebook.Runtime.ElixirStandalone do
+          receive do
+            {:DOWN, ^manager_ref, :process, _object, _reason} -> :ok
+          end
+      after
+        10_000 -> :timeout
+      end
+
+      # We explicitly halt at the end, just in case `System.no_halt(true)`
+      # is called within the runtime
+      System.halt()
+    end
+    |> Macro.to_string()
+    |> Base.encode64()
   end
 
   defp elixir_flags(node_name) do
     parent_name = node()
     parent_port = Livebook.EPMD.dist_port()
 
-    epmdless_flags =
-      if parent_port != 0 do
-        "-epmd_module Elixir.Livebook.EPMD -start_epmd false -erl_epmd_port 0 "
-      else
-        ""
-      end
-
     [
       "--erl",
-      # Minimize schedulers busy wait threshold,
-      # so that they go to sleep immediately after evaluation.
-      # Increase the default stack for dirty io threads (cuda requires it).
-      # Enable ANSI escape codes as we handle them with HTML.
-      # Disable stdin, so that the system process never tries to read terminal input.
+      # Note: keep these flags in sync with the remote runtime.
+      #
+      #   * minimize schedulers busy wait threshold, so that they go
+      #     to sleep immediately after evaluation
+      #
+      #   * increase the default stack for dirty IO threads, necessary
+      #     for CUDA
+      #
+      #   * enable ANSI escape codes as we handle them with HTML
+      #
+      #   * disable stdin, so that the system process never tries to
+      #     read terminal input
+      #
+      #   * specify a custom EPMD module and disable automatic EPMD
+      #     startup
+      #
       "+sbwt none +sbwtdcpu none +sbwtdio none +sssdio 128 -elixir ansi_enabled true -noinput " <>
-        epmdless_flags <>
-        "-livebook_parent #{parent_name} #{parent_port} -livebook_current #{node_name}",
-      # Add the location of Livebook.EPMD
+        "-epmd_module Elixir.Livebook.Runtime.EPMD",
+      # Add the location of Livebook.Runtime.EPMD
       "-pa",
-      Application.app_dir(:livebook, "priv/epmd"),
+      epmd_module_path!(),
       # Make the node hidden, so it doesn't automatically join the cluster
       "--hidden",
       # Use the cookie in Livebook
       "--cookie",
       Atom.to_string(Node.get_cookie()),
       "--eval",
-      @child_node_eval_string
+      "System.argv() |> hd() |> Base.decode64!() |> Code.eval_string()",
+      child_node_eval_string(node_name, parent_name, parent_port)
     ]
   end
+
+  defp epmd_module_path!() do
+    # We need to make the custom Livebook.Runtime.EPMD module available
+    # before the child node starts distrubtion. We persist the module
+    # into a temporary directory and add to the code paths. Note that
+    # we could persist it to priv/ at build time, however for Escript
+    # priv/ is packaged into the archive, so it is not accessible in
+    # the file system.
+
+    epmd_path = Path.join(Livebook.Config.tmp_path(), "epmd")
```

Comment:
Do we want to append the Livebook version on Livebook.Config dir or here? Otherwise if for some weird reason I am running two different versions, they will clash?

Reply:
Good call!